includes/rest-api/Controllers/Version2/class-wc-rest-products-v2-controller.php

```diff
@@ -589,86 +589,254 @@ protected function get_attributes( $product ) {
-			'external_url'          => $product->is_type( 'external' ) ? $product->get_product_url( $context ) : '',
-			'button_text'           => $product->is_type( 'external' ) ? $product->get_button_text( $context ) : '',
-			'tax_status'            => $product->get_tax_status( $context ),
-			'tax_class'             => $product->get_tax_class( $context ),
-			'manage_stock'          => $product->managing_stock(),
-			'stock_quantity'        => $product->get_stock_quantity( $context ),
-			'in_stock'              => $product->is_in_stock(),
-			'backorders'            => $product->get_backorders( $context ),
-			'backorders_allowed'    => $product->backorders_allowed(),
-			'backordered'           => $product->is_on_backorder(),
-			'sold_individually'     => $product->is_sold_individually(),
-			'weight'                => $product->get_weight( $context ),
-			'dimensions'            => array(
-				'length' => $product->get_length( $context ),
-				'width'  => $product->get_width( $context ),
-				'height' => $product->get_height( $context ),
-			),
-			'shipping_required'     => $product->needs_shipping(),
-			'shipping_taxable'      => $product->is_shipping_taxable(),
-			'shipping_class'        => $product->get_shipping_class(),
-			'shipping_class_id'     => $product->get_shipping_class_id( $context ),
-			'reviews_allowed'       => $product->get_reviews_allowed( $context ),
-			'average_rating'        => 'view' === $context ? wc_format_decimal( $product->get_average_rating(), 2 ) : $product->get_average_rating( $context ),
-			'rating_count'          => $product->get_rating_count(),
-			'related_ids'           => array_map( 'absint', array_values( wc_get_related_products( $product->get_id() ) ) ),
-			'upsell_ids'            => array_map( 'absint', $product->get_upsell_ids( $context ) ),
-			'cross_sell_ids'        => array_map( 'absint', $product->get_cross_sell_ids( $context ) ),
-			'parent_id'             => $product->get_parent_id( $context ),
-			'purchase_note'         => 'view' === $context ? wpautop( do_shortcode( wp_kses_post( $product->get_purchase_note() ) ) ) : $product->get_purchase_note( $context ),
-			'categories'            => $this->get_taxonomy_terms( $product ),
-			'tags'                  => $this->get_taxonomy_terms( $product, 'tag' ),
-			'images'                => $this->get_images( $product ),
-			'attributes'            => $this->get_attributes( $product ),
-			'default_attributes'    => $this->get_default_attributes( $product ),
-			'variations'            => array(),
-			'grouped_products'      => array(),
-			'menu_order'            => $product->get_menu_order( $context ),
-			'meta_data'             => $product->get_meta_data(),
+		$fields = array();
+		$request = func_get_arg( 2 );
+		if ( $request instanceof WP_REST_Request ) {
+			$fields = $this->get_fields_for_response( $request );
+		}
+		$base_data = array();
+		foreach ( $fields as $field ) {
+			switch ( $field ) {
```

Comment:
What do you think about defining `api_get_*` getter methods for these fields instead of having a `switch` statement?

Reply:
I was confused between these two approaches, `api_get_*` needs a lot more boiler plate than a switch as you see above in this file (which is not a problem on its own) but provides similar readibility. I am actually hoping to form an opinion and stick to only one approach based on this PR's review comments. Why do you think `api_get_*` is better than a switch construct?

Reply:
My initial resistance to the large switch statement was purely aesthetic, but I do think small getter functions for each field could improve readability and reduce maintenance burden. As more fields might get added, or perhaps the logic in the field retrieval needs to get more complicated - the switch statement (and therefore `get_formatted_item_data()` function) just grows larger.

Reply:
My suggestion is that we might settle on using a `switch` for now in favor of a more robust implementation in a future version of the API. 

My design preference would be for a `Transformer` of some sort that takes in a business model and returns a structured response to be used in the API. This is a nice pattern for abstracting away the presentation logic from the specific API endpoint and can even provide flexibility in having different responses for different versions of the API moving forward.